### PR TITLE
Build: Golang 1.20 upgrade.

### PR DESCRIPTION
**Pull request by gmalouf**
## Summary

Exactly what it sounds like; upgrade go-algorand to compile with Golang 1.20.5.

Note: Upon merge, developers need to re-run `./scripts/buildtools/install_buildtools.sh` to get updated versions of our buildtools.

## Test Plan

Existing unit/functional/integration tests all pass. Builds (on PR, nightly, etc) all function as intended.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -317,3 +317,5 @@ gopkg.in/yaml.v3 v3.0.1/go.mod h1:K4uyk7z7BCEPqu6E+C64Yfv1cQ7kz7rIZviUmN+EgEM=
 honnef.co/go/tools v0.0.0-20190102054323-c2f93a96b099/go.mod h1:rf3lG4BRIbNafJWhAfAdb/ePZxsR/4RtNHQocxwk9r4=
 pgregory.net/rapid v0.4.8 h1:d+5SGZWUbJPbl3ss6tmPFqnNeQR6VDOFly+eTjwPiEw=
 pgregory.net/rapid v0.4.8/go.mod h1:Z5PbWqjvWR1I3UGjvboUuan4fe4ZYEYNLNQLExzCoUs=
+pgregory.net/rapid v0.6.2 h1:ErW5sL+UKtfBfUTsWHDCoeB+eZKLKMxrSd1VJY6W4bw=
+pgregory.net/rapid v0.6.2/go.mod h1:PY5XlDGj0+V1FCq0o192FdRhpKHGTRIWBgqjDBTrq04=
```
[the remark this answers is not in the document]
Comment (gmalouf) [reply go.sum:319]: I think addressed, I typically use sanity which was broken on formatting first. Updated once I got through that (had PR marked draft while doing it).

File: .golangci.yml [hunk 2]
```diff
@@ -70,6 +70,60 @@ linters-settings:
...
+    # Default: false
+    enable-all-rules: false
+
+    # Sets the default failure confidence.
+    # This means that linting errors with less than 0.8 confidence will be ignored.
+    # Default: 0.8
+    confidence: 0.8
+
+    # In order to disable any default rules, you have to list out all rules you would like enabled.
+    # We explicitly disable (vs omitting) rules to make it clear how we are diverging.
```
[the remark this answers is not in the document]
Comment (gmalouf) [reply .golangci.yml:94]: I wasn't clear you could finetune revive's sub-rules in exclude-rules and saw multiple PR discussions laying out that going off the beaten path overrode golangci's default config for revive (requiring a full lay out like so).

**Comment by bbroder-algo** [comment]
it looks ok to me gary

**Review by excalq — APPROVED** [review]
Nice to see this! HyperFlow is assessing what corresponding changes are needed among Jenkinsfiles and Dockerfiles in the CI repo.

**Review by algobarb — COMMENTED** [review]
LGTM. CI-related changes, get_golang_version, and go.mod changes look good to me.

Other than formatting changes, I guess the other changes were also to support the new golang version that I saw in some files:
- Removing `.(uint64)` and `.(int)`
- Adding `[string]`

**Review by algobarb — APPROVED** [review]

**Pull request merged**